### ISSUE TITLE
network.host _[networkInterface]_ not working in Centos 6.5

### ISSUE THREAD

**Issue by jpcarey** [elastic/elasticsearch#17196]
<<< Update:
Just realized the machine that I was having issues with was actually running 2.1.1, and the others were 2.2.0.  Issue reproduces with 2.1.1, regardless of the OS.

I could not find an issue to link to, but some commit fixed this issue between 2.1.1 and 2.2.0.
>>>

**Elasticsearch version**: 2.2.0

**JVM version**: 1.8.0_45 (oracle)

**OS version**: Centos 6.5

The following configuration works fine under Ubuntu 14.04, but not under Centos 6.5:
```
network.host:
  - _eth0:ipv4_
  - _local:ipv4_

-----
publish_address {10.0.2.15:9300}, bound_addresses {127.0.0.1:9300}, {10.0.2.15:9300}
```

Same config, Centos 6.5 result: 
```
publish_address {127.0.0.1:9200}, bound_addresses {127.0.0.1:9200}, {[::1]:9200}
```

##### Other items that I tried (which **all** worked under Ubuntu 14.04):
- Works
```
network.host: "_eth0:ipv4_"

-----
publish_address {10.0.2.15:9300}, bound_addresses {10.0.2.15:9300}
```

- Not working:
```
network.host: ["_eth0:ipv4_", "_local_"]

-----
publish_address {127.0.0.1:9300}, bound_addresses {127.0.0.1:9300}, {[::1]:9300}
```

- Not working:
```
network.host:
 - "_eth0:ipv4_"

-----
publish_address {127.0.0.1:9300}, bound_addresses {127.0.0.1:9300}, {[::1]:9300}
```

- Not working:
```
network.host:
 - _eth0:ipv4_

-----
publish_address {127.0.0.1:9300}, bound_addresses {127.0.0.1:9300}, {[::1]:9300}
```

**Comment by jpcarey** [comment]
adding that it worked fine under Centos7 as well.

**Comment by jpcarey** [comment]
Just realized the machine that I was having issues with was actually running 2.1.1, and the others were 2.2.0.  Issue reproduces with 2.1.1, regardless of the OS.

I could not find an issue to link to, but some commit fixed this issue between 2.1.1 and 2.2.0.

**Comment by dadoonet** [comment]
I think you were looking for https://github.com/elastic/elasticsearch/pull/13954
